### ISSUE TITLE
Impulse101 multiplayer issue

### ISSUE THREAD

**Issue by Shurutsue** [naelstrof/KoboldKare#231]
Using impulse101 as a client (not the host) seems to cause issues - resulting in the client's kobold gaining excessive amounts of fat and energy, whereas stomach capacity and metabolism capacity stay the same.

The issue is most likely that it doesn't await the transfer of ownership and with the RPC call to increase the energy it'll mess things up? (and the genes not fully syncing prior to the other player regaining ownership(?))

Potential fix could be implementing a setGenesRPC method, which could be used.
Or to limit cheat-use exclusively to current "host".
Or to await the transfer of ownership and ensure the genes are synced prior to the other player regaining ownership through actions (like jumping?)

**Comment by naelstrof** [comment]
Yeah-- cheats need to be revisited and refactored. For whatever reason I made it so that it does a ping-pong with the host as a form of security. Though a malicious client wouldn't need to ping pong anyway so I might as well make cheats just check the room status and run the cheats directly if cheating is enabled.

Once it's refactored I imagine bugs like this would just disappear.
